pkg/tests/tasks/security/authentication/mtls_migration_test.go

```diff
@@ -16,134 +16,110 @@ package authentication
-				}
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, "foo", "bar", "legacy") // TODO: recreate all three namespaces with a single call to RecreateNamespace
+		})
+
+		app.InstallAndWaitReady(t,
+			app.Httpbin("foo"),
+			app.Httpbin("bar"),
+			app.Sleep("foo"),
+			app.Sleep("bar"),
+			app.SleepNoSidecar("legacy"))
+
+		fromNamespaces := []string{"foo", "bar", "legacy"}
+		toNamespaces := []string{"foo", "bar"}
+
+		t.LogStep("Check connectivity from namespaces foo, bar, and legacy to namespace foo and bar")
+		for _, from := range fromNamespaces {
+			for _, to := range toNamespaces {
+				assertConnectionSucessful(t, from, to)
 			}
 		}
-	})
 
-	t.Run("Security_authentication_globally_enable_mtls", func(t *testing.T) {
-		defer util.RecoverPanic(t)
-
-		log.Log.Info("Lock down to mutual TLS for the entire mesh")
-		util.KubeApplyContents(meshNamespace, util.RunTemplate(MeshPolicyStrictTemplate, smcp))
-		time.Sleep(time.Duration(30) * time.Second)
-
-		for _, from := range []string{"legacy"} {
-			for _, to := range []string{"foo", "bar"} {
-				sleepPod, err := util.GetPodName(from, "app=sleep")
-				util.Inspect(err, "Failed to get sleep pod name", "", t)
-				cmd := fmt.Sprintf(`curl http://httpbin.%s:8000/ip -s -o /dev/null -w "sleep.%s to httpbin.%s: %%{http_code}"`, to, from, to)
-				msg, err := util.PodExec(from, sleepPod, "sleep", cmd, true)
-				if from == "legacy" && to == "foo" {
-					if err != nil {
-						log.Log.Infof("Expected sleep.legacy to httpbin.foo fails: %v", err)
-					} else {
-						t.Errorf("Expected sleep.legacy to httpbin.foo fails; Got unexpected response: %s", msg)
-						log.Log.Errorf("Expected sleep.legacy to httpbin.foo fails; Got unexpected response: %s", msg)
+		t.NewSubTest("mTLS enabled in foo").Run(func(t test.TestHelper) {
+			t.LogStep("Apply strict mTLS in namespace foo")
+			oc.ApplyString(t, "foo", NamespacePolicyStrict)
+
+			t.LogStep("Check connectivity from legacy to foo (expect failure) and bar (expect success)")
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				for _, from := range fromNamespaces {
+					for _, to := range toNamespaces {
+						if from == "legacy" && to == "foo" {
+							assertConnectionFailure(t, from, to)
+						} else {
+							assertConnectionSucessful(t, from, to)
+						}
 					}
-					continue
 				}
-				if from == "legacy" && to == "bar" {
-					if err != nil {
-						log.Log.Infof("Expected sleep.legacy to httpbin.bar fails: %v", err)
-					} else {
-						t.Errorf("Expected sleep.legacy to httpbin.bar fails; Got unexpected response: %s", msg)
-						log.Log.Errorf("Expected sleep.legacy to httpbin.bar fails; Got unexpected response: %s", msg)
+			})
+		})
+
+		t.NewSubTest("mTLS enabled globally").Run(func(t test.TestHelper) {
+			t.LogStep("Apply strict mTLS for the entire mesh")
+			oc.ApplyTemplate(t, meshNamespace, MeshPolicyStrictTemplate, smcp)
+			t.Cleanup(func() {
+				oc.DeleteFromTemplate(t, meshNamespace, MeshPolicyStrictTemplate, smcp)
```

Comment:
Is there a way we can check the deletion of a mesh wide policy completion ?
If that deletion takes time to complete, it may affect the execution of running the next test case when we trigger a full run.

Reply:
There's no way to do this and have it work correctly 100% of the time. However, if all the tests use retries, as they should, this won't be a problem.